packages/frontend/amp/components/ReaderRevenueButton.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import { textSans } from '@guardian/pasteup/typography';
+import ArrowRight from '@guardian/pasteup/icons/arrow-right.svg';
+import { palette } from '@guardian/pasteup/palette';
+
+const supportStyles = css`
+    align-self: flex-start;
+    background-color: ${palette.highlight.main};
+    border-radius: 20px;
+    display: flex;
+    align-items: center;
+    padding: 0 15px;
+    min-height: 30px;
+`;
+
+const supportHeaderStyles = css`
+    ${supportStyles}
+    justify-content: center;
+    margin-top: 20px;
+    margin-left: 20px;
+`;
+
+const supportFooterStyles = css`
+    ${supportStyles}
+    margin-bottom: 6px;
+`;
+
+const supportLinkStyles = css`
+    position: relative;
+    color: ${palette.neutral[7]};
+    ${textSans(5)};
+    display: block;
+    line-height: 20px;
+    text-decoration: none;
+    padding-right: 20px;
+    width: 100%;
+
+    svg {
+        position: absolute;
+        top: -6px;
+    }
+`;
+
+const rightAlignedIcon = css`
+    position: absolute;
+    height: 20px;
+    width: 20px;
+    right: 0;
+    top: 0;
+`;
+
+export const ReaderRevenueButton: React.SFC<{
+    nav: NavType;
+    linkLabel: string;
+    rrLink: ReaderRevenueLinkNames;
+    rrCategory: ReaderRevenueCategoryNames;
+    rightAlignIcon?: boolean;
+}> = ({ nav, linkLabel, rrLink, rrCategory, rightAlignIcon }) => {
+    const url = nav.readerRevenueLinks[rrLink][rrCategory];
+
+    if (url === '') {
+        return null;
+    }
+
+    return (
+        <div
+            className={
+                rrLink === 'ampHeader'
+                    ? supportHeaderStyles
+                    : supportFooterStyles
+            }
```

Comment:
We have been using [`cx`](https://github.com/guardian/dotcom-rendering/blob/master/docs/contributing/code-style.md#prefer-cx-for-style-composition) to handle classname logic. Although this is going away in Emotion 10.

Reply:
Ok, I'll change this to use the recommend alternative here https://github.com/emotion-js/emotion/blob/master/docs/migrating-to-emotion-10.md#thinking

Reply:
I think we should have a chat about this. I really don't like emotion's decision to customise JSX for version 10. It couples Emotion to JSX which is really unhelpful (they do it for various optimisations, but it has a lot of downsides too).

It's still new to me (so I need to have a better look), but at the moment my preference would be to keep us using classnames directly, rather than their new and 'improved' approach.

What do others think? @GHaberis @SiAdcock

Reply:
Yeah, I think we should discuss this more as it's quite the design decision now I read up on it a little more. There are interesting consequences and arguments either way.

For now, I'll follow convention and use `cx` in the thought that if/when we migrate, we can move as part of that.

Reply:
Though `It forces people to think in terms of styling elements instead of creating class names` might actually be something we want. Why did we go with CSS class name styling and not go *FULL CSS-in-JS*?

Reply:
There are some benefits to staying with class names. The main one is to avoid coupling the CSS approach too tightly to the specific framework (React) we were using.

But there are some other big cons to styled components. E.g. consider:

    <div>
        <ul>
             <li>
                 <a href=..>foo</a>
             </li>
        </ul>
    </div>

vs something like:

    <MyDiv>
        <MyList>
             <ListItem1>
                 <ALink href=..>foo</a>
             </ListItem1>
        </MyList>
    </MyDiv>

In the second case:

* the actual eventual markup is not clear at all
* you are forced to come up with good names for your elements (and naming things well is notoriously a difficult thing)

Lastly, we went for tagged template literals rather than object styles, to:

* limit power (objects open up all of JS which can lead to confusing code for limited benefits in this specific case)
* be familiar to non-JS devs
* allow easy copy-paste between code and browser etc.

https://docs.google.com/document/d/145--cCors1_RhEGRkmuSCYYqcoYO_xkYlKBtzRD-P6Y/edit# has some of our thinking at the time, but it doesn't really capture the full discussion - as we spent a bit of meeting time discussing the doc and refining our thoughts after writing this.